File: apps/android/core/testing/src/main/kotlin/app/gauja/core/testing/FakeClock.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.testing

import java.time.Clock
import java.time.Instant
import java.time.ZoneId
import java.time.ZoneOffset

class FakeClock(var now: Instant = Instant.parse("2026-01-01T00:00:00Z")) : Clock() {
    override fun instant(): Instant = now

    override fun getZone(): ZoneId = ZoneOffset.UTC

    override fun withZone(zone: ZoneId): Clock = fixed(now, zone)
}

File: apps/ios/Packages/Testing/Sources/GaujaTesting/FakeClock.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
import Common
import Foundation

public struct FakeClock: Sendable {
    public nonisolated let clock: WallClock
    public nonisolated init(now: Date = Date(timeIntervalSince1970: 1_767_225_600)) { clock = WallClock(now: { now }) }
}
